modules/channel-vonage/src/renderers/template.ts

```diff
@@ -0,0 +1,34 @@
+import { ChannelRenderer } from 'common/channel'
+import { CHANNEL_NAME } from '../backend/client'
+import { VonageContext } from '../backend/typings'
+
+export class VonageTemplateRenderer implements ChannelRenderer<VonageContext> {
+  get channel(): string {
+    return CHANNEL_NAME
+  }
+
+  get priority(): number {
+    return 0
+  }
+
+  get id(): string {
+    return VonageTemplateRenderer.name
+  }
+
+  handles(context: VonageContext): boolean {
+    return context.payload.type === 'vonage-template'
+  }
+
+  async render(context: VonageContext) {
+    const payload = context.payload
+
+    context.messages.push({
+      type: 'template',
+      text: undefined,
+      template: {
+        name: `${payload.namespace}:${payload.name}`,
+        parameters: payload.parameters
+      }
```

Comment:
Continuing on my other comment, we could for example do as little transformation as this :

```suggestion
      template: context.payload.template
```